thrust/cmake/README_CMake.md

```diff
@@ -0,0 +1,211 @@
+# Using Thrust with CMake
+
+Thrust provides configuration files that simplify the process of using Thrust
+from other CMake projects. Requirements:
+
+- Thrust >= 1.9.10
```

Comment:
Please add a section with instructions of how to find the existing broken `FindThrust.cmake` with the subset of changes in this patch for projects using it, but also wanting to support older Thrusts, to still work.

Reply:
@griwes Added the following. Resolve this comment if it looks good to you.

## Fixing Legacy FindThrust.cmake

A community-created `FindThrust.cmake` module exists and is necessary to find
Thrust installations prior to Thrust 1.9.10. Its usage is discouraged whenever
possible and the config files in this directory should be strongly preferred.
However, projects that need to support old version of Thrust may still need to
use this find module with pre-1.9.10 installations.

One flavor of this find module has a version parsing bug. Projects that
rely on `FindThrust.cmake` should patch their copies as follows.

Replace:

```cmake
string( REGEX MATCH "^[0-9]" major ${version} )
string( REGEX REPLACE "^${major}00" "" version "${version}" )
string( REGEX MATCH "^[0-9]" minor ${version} )
string( REGEX REPLACE "^${minor}0" "" version "${version}" )
```

with:

```cmake
math(EXPR major "${version} / 100000")
math(EXPR minor "(${version} / 100) % 1000")
math(EXPR version "${version} % 100")
```